src/test/optimizer/strategy/between_composition_rule_test.cpp

```diff
@@ -0,0 +1,256 @@
+  // clang-format on
+
+  const auto result_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
+
+  EXPECT_LQP_EQ(result_lqp, expected_lqp);
+}
+
+TEST_F(BetweenCompositionTest, ImpossibleColumnRange) {
+  // clang-format off
+  const auto input_lqp =
+  PredicateNode::make(greater_than_equals_(_column_a, 300),
+    PredicateNode::make(less_than_equals_(_column_a, 200),
+      _node));
+
+  const auto expected_lqp = PredicateNode::make(between_(_column_a, 300, 200), _node);
+  // clang-format on
+
+  const auto result_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
+
+  EXPECT_LQP_EQ(result_lqp, expected_lqp);
+}
+
+TEST_F(BetweenCompositionTest, OrStatement) {
+  // clang-format off
+  const auto input_lqp = PredicateNode::make(or_(
+    greater_than_equals_(_column_a, 200),
+    less_than_equals_(_column_a, 300)),
+    _node);
+
+  const auto expected_lqp = input_lqp;
+  // clang-format on
+
+  const auto result_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
+
+  EXPECT_LQP_EQ(result_lqp, expected_lqp);
+}
+
+TEST_F(BetweenCompositionTest, LeftExclusive) {
+  // clang-format off
+  const auto input_lqp =
+  PredicateNode::make(greater_than_(_column_a, 200),
+    PredicateNode::make(less_than_equals_(_column_a, 300),
+      _node));
+
+  const auto expected_lqp =
+  PredicateNode::make(
+    std::make_shared<BetweenExpression>(
+      _column_a.original_node()->column_expressions()[0],
+      value_(200),
+      value_(300),
+      false,
+      true),
+    _node);
+  // clang-format on
+
+  const auto result_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
+
+  EXPECT_LQP_EQ(result_lqp, expected_lqp);
+}
+
+TEST_F(BetweenCompositionTest, RightExclusive) {
+  // clang-format off
+  const auto input_lqp =
+  PredicateNode::make(greater_than_equals_(_column_a, 200),
+    PredicateNode::make(less_than_(_column_a, 300),
+      _node));
+
+  const auto expected_lqp =
+  PredicateNode::make(
+    std::make_shared<BetweenExpression>(
+      _column_a.original_node()->column_expressions()[0],
+      value_(200),
+      value_(300),
+      true,
+      false),
+    _node);
+  // clang-format on
+
+  const auto result_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
+
+  EXPECT_LQP_EQ(result_lqp, expected_lqp);
+}
+
+TEST_F(BetweenCompositionTest, BothExclusive) {
+  // clang-format off
+  const auto input_lqp =
+  PredicateNode::make(greater_than_(_column_a, 200),
+    PredicateNode::make(less_than_(_column_a, 300),
+      _node));
+
+  const auto expected_lqp =
+  PredicateNode::make(
+    std::make_shared<BetweenExpression>(
+      _column_a.original_node()->column_expressions()[0],
+      value_(200),
+      value_(300),
+      false,
+      false),
+    _node);
+  // clang-format on
+
+  const auto result_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
+
+  EXPECT_LQP_EQ(result_lqp, expected_lqp);
+}
+
+TEST_F(BetweenCompositionTest, FindOptimalBetween) {
+  // clang-format off
+  const auto input_lqp =
```

Comment:
Does it make sense to mix in some `greater_than` without the `equals` in here?

Reply:
I will add another test for this case

Reply:
is it resolved?